File: backend/Models/Organization.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QM_AI.API.Models;

/// <summary>
/// 企业组织层级（自引用树形结构）
/// </summary>
[Table("organizations")]
public class Organization
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("id")]
    public long Id { get; set; }

    /// <summary>组织编码</summary>
    [Required]
    [MaxLength(50)]
    [Column("code")]
    public string Code { get; set; } = string.Empty;

    /// <summary>组织名称</summary>
    [Required]
    [MaxLength(200)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    /// <summary>层级: group/company/workshop/line</summary>
    [Required]
    [MaxLength(20)]
    [Column("level")]
    public string Level { get; set; } = string.Empty;

    /// <summary>父级组织ID</summary>
    [Column("parent_id")]
    public long? ParentId { get; set; }

    /// <summary>排序号</summary>
    [Column("sort_order")]
    public int SortOrder { get; set; } = 0;

    /// <summary>是否启用</summary>
    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    /// <summary>位置/地址</summary>
    [MaxLength(500)]
    [Column("location")]
    public string? Location { get; set; }

    /// <summary>联系人信息 (JSON)</summary>
    [Column("contact")]
    public string? Contact { get; set; }

    /// <summary>描述</summary>
    [Column("description")]
    public string? Description { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>创建人</summary>
    [Column("created_by")]
    public long? CreatedBy { get; set; }

    // Navigation properties
    [ForeignKey(nameof(ParentId))]
    public Organization? Parent { get; set; }

    public ICollection<Organization> Children { get; set; } = new List<Organization>();
}

File: frontend/src/types/organization.ts
// ─── Organization ─────────────────────────────────────────
export interface OrganizationTreeNode {
  id: number
  code: string
  name: string
  level: 'group' | 'company' | 'workshop' | 'line'
  parentId: number | null
  sortOrder: number
  isActive: boolean
  location?: string
  description?: string
  createdBy?: number
  childCount: number
  children: OrganizationTreeNode[]
}

export interface Organization {
  id: number
  code: string
  name: string
  level: string
  parentId: number | null
  parentName?: string
  sortOrder: number
  isActive: boolean
  location?: string
  description?: string
  createdAt: string
}

export interface OrganizationDetail {
  id: number
  code: string
  name: string
  level: string
  parentId: number | null
  parentName?: string
  sortOrder: number
  isActive: boolean
  location?: string
  contact?: string
  description?: string
  createdBy?: number
  createdAt: string
  updatedAt: string
}

export interface CreateOrganization {
  code: string
  name: string
  level: string
  parentId?: number | null
  sortOrder?: number
  location?: string
  contact?: string
  description?: string
  createdBy?: number
}

export interface UpdateOrganization {
  code: string
  name: string
  parentId?: number | null
  sortOrder?: number
  isActive?: boolean
  location?: string
  contact?: string
  description?: string
}

// 层级配置常量
export const LEVEL_CONFIG: Record<string, { label: string; color: string }> = {
  group: { label: '集团', color: '#8B5CF6' },
  company: { label: '公司', color: '#409EFF' },
  workshop: { label: '车间', color: '#E6A23C' },
  line: { label: '产线', color: '#67C23A' },
}

export const NEXT_LEVEL: Record<string, string> = {
  group: 'company',
  company: 'workshop',
  workshop: 'line',
  line: '',
}

export const LEVEL_ORDER: Record<string, number> = {
  group: 0,
  company: 1,
  workshop: 2,
  line: 3,
}
